src/k8s-configuration/azext_k8s_configuration/action.py

```diff
@@ -0,0 +1,68 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=protected-access
+
+import argparse
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from .vendored_sdks.v2021_11_01_preview.models import KustomizationDefinition, DependsOnDefinition
+from .validators import validate_kustomization
+from . import consts
+from .utils import parse_dependencies, parse_duration
+
+
+class InternalKustomizationDefinition(KustomizationDefinition):
```

Comment:
What is the purpose of these two classes in action.py?

Reply:
Since I can only pass a single object through the `action.py` I need an object which contains the name, `KustomizationDefinition` doesn't contain the name of the kustomization since we send the object off like this

```
{'nameOfKustomization': KustomizationDefinition}
```